src/pool/srv_pool.c

```diff
@@ -6186,11 +6186,20 @@ ds_pool_upgrade_if_needed(uuid_t pool_uuid, struct rsvc_hint *po_hint,
 		switch (upgrade_status) {
 		case DAOS_UPGRADE_STATUS_NOT_STARTED:
 		case DAOS_UPGRADE_STATUS_COMPLETED:
-			if ((upgrade_global_ver < DAOS_POOL_GLOBAL_VERSION &&
-			     dmg_upgrade_cmd) || DAOS_FAIL_CHECK(DAOS_FORCE_OBJ_UPGRADE))
+			if (DAOS_FAIL_CHECK(DAOS_FORCE_OBJ_UPGRADE)) {
 				D_GOTO(out_upgrade, rc = 0);
-			else
+			} else if (upgrade_global_ver < DAOS_POOL_GLOBAL_VERSION &&
+				   dmg_upgrade_cmd) {
+				if (DAOS_POOL_GLOBAL_VERSION - upgrade_global_ver == 1)
+					D_GOTO(out_upgrade, rc = 0);
+				D_ERROR(DF_UUID ": upgrading pool %u -> %u\n is unsupported"
```

Comment:
No change requested, just discussion. To follow this suggestion, one would have to downgrade the DAOS software version to get one whose DAOS_POOL_GLOBAL_VERSION matches the upgrade_global_ver+1 is that right?

If this is the case, could the message say "please downgrade DAOS and then up[grade pool to %u firstly"?

Reply:
Yup, that looks better, I could change that if PR need be refreshed for any reason.

Reply:
Also, probably a wild edge case. But what happens if someone upgrades the servers, creates a **_new_** pool at the current version, and then downgrades the servers? Is that new pool accessible again after the servers are updated?

Reply:
In that case, it will raise a RAS event and pool will be refused to start. And pool will be accessible again once the server are updated.